src/python/pants/backend/codegen/grpcio/python/grpcio_run.py

```diff
@@ -45,6 +47,11 @@ def execute_codegen(self, target, target_workdir):
       if exit_code != 0:
         raise TaskError('{} ... exited non-zero ({}).'.format(cmdline, exit_code),
                         exit_code=exit_code)
+      # Create __init__.py in each subdirectory of the target directory so that setup_py recognizes
+      # them as modules.
+      for root, dirs, _ in safe_walk(target_workdir):
```

Comment:
To add one more change...turns out that we actually don't need to use `safe_walk` anymore! This was necessary in Python 2 to ensure that Unicode paths are handled correctly. But as of last week, Pants now requires Python 3 to run so this is no longer a concern. 

(Fun fact: this PR will be our first usage of `pathlib.Path` because Python-3 only is so new!)

You can replace this with simply `os.walk`. I'll make a PR right now to remove `safe_walk()` from our codebase.

Reply:
Oof, never mind. I spoke too soon here. Turns out that even still in Python 3, if you pass a byte string to `os.walk`, it will use bytes for everything else.

Even though I'm confident here that `target_workdir` is `str` (i.e. unicode), I'd prefer that we keep `safe_walk` as is. We'll change this usage in a followup that decides how we should handle this API, likely choosing one of the two below options:
1) Remove the function and rely on people using `os.walk` properly.
2) Rewrite `safe_walk` to assert that the argument is `str` and raise an exception if it's `bytes`.

For this PR, we should keep what you have.